libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/WayNameView.java

```diff
@@ -32,8 +34,23 @@ public WayNameView(@NonNull Context context, @Nullable AttributeSet attrs, int d
     initialize();
   }
 
+  @Override
+  public void setVisibility(int visibility) {
+    if (visibility == VISIBLE) {
+      visible = true;
```

Comment:
if a caller explicitly call `setVisibility` to **VISIBLE** or call `updateVisibility(true)`, then we remember it internally. But will set the visibility base on the wayNameText.